### PR TITLE
Make HLE project AOT friendly

### PR DESCRIPTION
**Pull request by emmauss**
We make heavy use of Reflection apis in the HLE project, esp. the Service api, making it entirely incompatible with native aot.

This PR removes all aot-incompatible usages of Reflection in the HLE and Graphics project. The changes are;
1. Add an IpcServiceGenerator to generate paths to the constructors of all IpcServices. This doesn't create paths to the Ipc methods, and isn't a replacement for the Horizon project. As such, an rd.xml file must be included with any executable project being published with native aot.
2. Remove SizeCalculator class, and instead use Marshal api for related structs.
3. Simplify AppletManager.

There shouldn't be any noticeable change with this pr.

### CODE REVIEW

**Comment by gdkchan** [comment]
> As such, an rd.xml file must be included with any executable project being published with native aot.

Is it supposed to be included with this change? I'm not seeing it here.

File: src/Ryujinx.Graphics.Device/DeviceState.cs [hunk 1]
```diff
@@ -39,7 +39,10 @@ public DeviceState(IReadOnlyDictionary<string, RwCallback> callbacks = null, Act
             {
                 var field = fields[fieldIndex];
 
-                int sizeOfField = SizeCalculator.SizeOf(field.FieldType);
```
Comment (gdkchan) [inline src/Ryujinx.Graphics.Device/DeviceState.cs:42]: We should replace this with https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.runtimehelpers.sizeof?view=net-9.0 once we migrate to .NET 9.
This is fine as a stopgap solution.

File: src/Ryujinx.HLE/Ryujinx.HLE.csproj [hunk 2]
```diff
@@ -11,7 +11,9 @@
     <ProjectReference Include="..\Ryujinx.Graphics.Host1x\Ryujinx.Graphics.Host1x.csproj" />
     <ProjectReference Include="..\Ryujinx.Graphics.Nvdec\Ryujinx.Graphics.Nvdec.csproj" />
     <ProjectReference Include="..\Ryujinx.Graphics.Vic\Ryujinx.Graphics.Vic.csproj" />
+    <ProjectReference Include="..\Ryujinx.HLE.Generators\Ryujinx.HLE.Generators.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false"/>
     <ProjectReference Include="..\Ryujinx.Horizon.Common\Ryujinx.Horizon.Common.csproj" />
+    <ProjectReference Include="..\Ryujinx.Horizon.Generators\Ryujinx.Horizon.Generators.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
```
Comment (gdkchan) [inline src/Ryujinx.HLE/Ryujinx.HLE.csproj:16]: Why was that included here?

**Comment by emmauss** [comment]
> > As such, an rd.xml file must be included with any executable project being published with native aot.
> 
> Is it supposed to be included with this change? I'm not seeing it here.

No. It will be included when native aot is enabled on the main project, as it's more suitable there

File: src/Ryujinx.HLE/Ryujinx.HLE.csproj (hunk 2 above)
Comment (emmauss) [reply src/Ryujinx.HLE/Ryujinx.HLE.csproj]: Removed

File: src/Ryujinx.HLE.Generators/IpcServiceGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+namespace Ryujinx.HLE.Generators
+{
+    [Generator]
+    public class IpcServiceGenerator : ISourceGenerator
+    {
+        public void Execute(GeneratorExecutionContext context)
+        {
+            var syntaxReceiver = (ServiceSyntaxReceiver)context.SyntaxReceiver;
+            CodeGenerator generator = new CodeGenerator();
+            generator = new CodeGenerator();
```
Comment (GreemDev) [inline src/Ryujinx.HLE.Generators/IpcServiceGenerator.cs:15]: I'm guessing this double assignment is an accident

Comment (emmauss) [reply src/Ryujinx.HLE.Generators/IpcServiceGenerator.cs]: Yes. Removed.

**Review by riperiperi — APPROVED** [review]

**Review by gdkchan — APPROVED** [review]
lgtm, thanks. No need to be strict with the code style/implementation here since this is supposed to be a temporary solution (until we migrate all services to new IPC). Use of `Marshal.OffsetOf` is "incorrect" (should not be a problem with our current types tho), but we should be able to migrate to `RuntimeHelpers.SizeOf` once we migrate to .NET 9.
